### PR TITLE
Added --enable-azure-rbac flag in ManagedClusterAadProfile (#2)

### PR DESCRIPTION
**Pull request by Anumita**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by Anumita** [comment]
@weinong can you review the PR?

**Comment by yonzhan** [comment]
AKS

**Comment by gtxistxgao** [comment]
Seems that 2020-06-01 has not yet been published. https://msazure.visualstudio.com/CloudNativeCompute/_git/manifests?path=%2Frp%2Fglobal.json&version=GBmaster&line=2067&lineEnd=2068&lineStartColumn=1&lineEndColumn=25&lineStyle=plain

File: src/aks-preview/azext_aks_preview/_help.py [hunk 1]
```diff
@@ -70,6 +70,9 @@
         - name: --enable-aad
           type: bool
           short-summary: Enable managed AAD feature for cluster.
+        - name: --aad-enable-azure-rbac
```
Comment (weinong) [inline src/aks-preview/azext_aks_preview/_help.py:73]: should be --enable-azure-rbac?

Comment (Anumita) [reply src/aks-preview/azext_aks_preview/_help.py:73]: As it is aad flag, i used aad as a prefix following the notation as used in aad-admin-group-object-ids and aad-tenant-id flags

Comment (Anumita) [reply src/aks-preview/azext_aks_preview/_help.py:73]: is that fine or should i use enable-azure-rbac?

Comment (weinong) [reply src/aks-preview/azext_aks_preview/_help.py:73]: i'd think so. but please confirm with Jorge/Justin/Shashank

Comment (Anumita) [reply src/aks-preview/azext_aks_preview/_help.py:73]: @shashankbarsin, @palma21 , @jluk, what do think?

Comment (palma21) [reply src/aks-preview/azext_aks_preview/_help.py:73]: Yes, enable-azure-rbac is good

Comment (Anumita) [reply src/aks-preview/azext_aks_preview/_help.py]: Changed to --enable-azure-rbac as discussed

File: src/aks-preview/azext_aks_preview/custom.py [hunk 2]
```diff
@@ -49,22 +49,22 @@
...
+from .vendored_sdks.azure_mgmt_preview_aks.v2020_06_01.models import ManagedCluster
+from .vendored_sdks.azure_mgmt_preview_aks.v2020_06_01.models import ManagedClusterAADProfile
+from .vendored_sdks.azure_mgmt_preview_aks.v2020_06_01.models import ManagedClusterAddonProfile
+from .vendored_sdks.azure_mgmt_preview_aks.v2020_06_01.models import ManagedClusterAgentPoolProfile
+from .vendored_sdks.azure_mgmt_preview_aks.v2020_06_01.models import AgentPool
+from .vendored_sdks.azure_mgmt_preview_aks.v2020_06_01.models import AgentPoolUpgradeSettings
+from .vendored_sdks.azure_mgmt_preview_aks.v2020_06_01.models import ContainerServiceStorageProfileTypes
+from .vendored_sdks.azure_mgmt_preview_aks.v2020_06_01.models import ManagedClusterIdentity
+from .vendored_sdks.azure_mgmt_preview_aks.v2020_06_01.models import ManagedClusterAPIServerAccessProfile
+from .vendored_sdks.azure_mgmt_preview_aks.v2020_06_01.models import ManagedClusterSKU
```
Comment (qwordy) [inline src/aks-preview/azext_aks_preview/custom.py:67]: Can be combined. Then you only need to modify one place.
from a import (A, B, C)

File: src/aks-preview/azext_aks_preview/_help.py [hunk 3]
```diff
@@ -70,6 +70,9 @@
         - name: --enable-aad
           type: bool
           short-summary: Enable managed AAD feature for cluster.
+        - name: --enable-azure-rbac
```
Comment (yungezz) [inline src/aks-preview/azext_aks_preview/_help.py:73]: could you pls use `enable-rbac` as param name to consistent with existing commands?

File: src/aks-preview/azext_aks_preview/__init__.py [hunk 4]
```diff
@@ -16,7 +16,7 @@ class ContainerServiceCommandsLoader(AzCommandsLoader):
 
     def __init__(self, cli_ctx=None):
         from azure.cli.core.commands import CliCommandType
-        register_resource_type('latest', CUSTOM_MGMT_AKS_PREVIEW, '2020-04-01')
+        register_resource_type('latest', CUSTOM_MGMT_AKS_PREVIEW, '2020-06-01')
```
Comment (yungezz) [inline src/aks-preview/azext_aks_preview/__init__.py:19]: pls make sure you do full LIVE test regression when bumping API version

File: src/aks-preview/azext_aks_preview/_help.py (hunk 3 above)
Comment (Anumita) [reply src/aks-preview/azext_aks_preview/_help.py:73]: enable-rbac has a different functionality then enable-azure-rbac though and in the api we have added enableAzureRBAC as the param. Also, enable-azure-rbac will only be for aad v2 clusters at the time of creation.

File: src/aks-preview/azext_aks_preview/__init__.py (hunk 4 above)
Comment (Anumita) [reply src/aks-preview/azext_aks_preview/__init__.py:19]: could you elaborate a little? not sure what i need to do as LIVE test regression

Comment (yungezz) [reply src/aks-preview/azext_aks_preview/__init__.py:19]: this change bumping major API version. there's could be breaking change cross different API versions. Run existing tests against real AKS could help to find out these change to make sure no regression in existing CLI commands.

File: src/aks-preview/azext_aks_preview/_help.py (hunk 3 above)
Comment (yungezz) [reply src/aks-preview/azext_aks_preview/_help.py:73]: thanks. what's difference between `enable-rbac` and `enable-azure-rbac`?

Comment (Anumita) [reply src/aks-preview/azext_aks_preview/_help.py:73]: enable-rbac enables K8s RBAC mode for authorization where as enable-azure-rbac will enable webhook mode in K8s for authorization, the webhook server that we have deployed using guard which will receive the authorization requests from K8s and convert them into appropriate checkaccess calls to send to ARM and then return the result whether the user is authorized to perform the selected action or not

File: src/aks-preview/azext_aks_preview/__init__.py (hunk 4 above)
Comment (Anumita) [reply src/aks-preview/azext_aks_preview/__init__.py:19]: got it. will run the existing test cli commands and check if everything is creating properly

**Comment by arrownj** [comment]
Hi @bingosummer , could you please help resolve the merge conflict ?

File: src/aks-preview/HISTORY.md [hunk 5]
```diff
@@ -12,6 +12,7 @@ Release History
 
 0.4.52
 +++++
+* Add "--enable-azure-rbac" for enabling Azure RBAC for Kubernetes authorization
```
Comment (bingosummer) [inline src/aks-preview/HISTORY.md:15]: you may need to add a new version in HISTORY.md and https://github.com/Azure/azure-cli-extensions/blob/master/src/aks-preview/setup.py.

File: src/aks-preview/azext_aks_preview/vendored_sdks/azure_mgmt_preview_aks/v2020_06_01/models/_models.py [hunk 6]
```diff
@@ -1494,7 +1492,7 @@ class ManagedClusterAgentPoolProfile(ManagedClusterAgentPoolProfileProperties):
     :param max_pods: Maximum number of pods that can run on a node.
     :type max_pods: int
     :param os_type: OsType to be used to specify os type. Choose from Linux
-     and Windows. Default to Linux. Possible values include: 'Linux',
+     and Windows. Default to Linux. Possible values include: 'Linu
```
Comment (bingosummer) [inline src/aks-preview/azext_aks_preview/vendored_sdks/azure_mgmt_preview_aks/v2020_06_01/models/_models.py:1495]: typo?

File: src/aks-preview/azext_aks_preview/tests/latest/recordings/test_aks_create_and_update_with_managed_aad_enable_azure_rbac.yaml [hunk 7]
```diff
@@ -0,0 +1,561 @@
...
+      message: OK
+- request:
+    body: '{"location": "westus2", "properties": {"kubernetesVersion": "", "dnsPrefix":
+      "cliakstest-clitestvouk66jlj-c10894", "agentPoolProfiles": [{"count": 1, "vmSize":
+      "Standard_D2s_v3", "osType": "Linux", "type": "AvailabilitySet", "mode": "System",
+      "enableNodePublicIP": false, "scaleSetPriority": "Regular", "scaleSetEvictionPolicy":
+      "Delete", "name": "nodepool1"}], "linuxProfile": {"adminUsername": "azureuser",
+      "ssh": {"publicKeys": [{"keyData": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDiOFgDVZSitF+jMxvBpJmYO8ReNVZMG+n996WvKxFfaR9903Zykxrvu7RUqdpZZ0hzzueeLJPJpiGxvk8ur4vnkXJkX7H8fvZZw6EI5PhuD76xXXRdXp9I2dH91oTqJ0Ne/s01K2PcJC+SzkU+icGLsIqP47QVoBL+H1nMapzcZVYYXzPd2u1bJZbGgJZSxEOJenybNY/1qhJGIEhMqLPXcz8QDuqLMrz3WMGLmfwv5SMTzd9ejfRAJ85RXeciTSjUtT37HUDafiql3JxhrrbUkgyuoVMztGp6DvTHb0XKDS06su1jBxQpGuquCKcpJNIa7Z3pJmWSM/b7VahsHzWP
+      pmiller@gmail.com\n"}]}}, "servicePrincipalProfile": {"clientId": "709e9be8-2b36-441f-81b5-16bd257fb5cb",
+      "secret": "b39ff07ddb2703825634$"}, "addonProfiles": {}, "enableRBAC": true,
```
Comment (bingosummer) [inline src/aks-preview/azext_aks_preview/tests/latest/recordings/test_aks_create_and_update_with_managed_aad_enable_azure_rbac.yaml:56]: do we need to redact the credentials?

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 8]
```diff
@@ -109,6 +109,26 @@ def test_aks_create_nonaad_and_update_with_managed_aad(self, resource_group, res
...
+    @ResourceGroupPreparer(random_name_length=17, name_prefix='clitest', location='westus2')
+    def test_aks_create_and_update_with_managed_aad_enable_azure_rbac(self, resource_group, resource_group_location):
+        aks_name = self.create_random_name('cliakstest', 16)
+        self.kwargs.update({
+            'resource_group': resource_group,
+            'name': aks_name
+        })
+
+        create_cmd = 'aks create --resource-group={resource_group} --name={name} ' \
+                     '--vm-set-type  AvailabilitySet -c 1 ' \
```
Comment (bingosummer) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:123]: shall we use VMSS as `--vm-set-type`?

Comment (jluk) [reply src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:123]: yes please default to VMSS, always in CLI

File: src/aks-preview/azext_aks_preview/tests/latest/recordings/test_aks_create_and_update_with_managed_aad_enable_azure_rbac.yaml (hunk 7 above)
Comment (Anumita) [reply src/aks-preview/azext_aks_preview/tests/latest/recordings/test_aks_create_and_update_with_managed_aad_enable_azure_rbac.yaml:56]: we have the same in the other recordings

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py (hunk 8 above)
Comment (Anumita) [reply src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py]: done

File: src/aks-preview/HISTORY.md (hunk 5 above)
Comment (Anumita) [reply src/aks-preview/HISTORY.md]: added

File: src/aks-preview/azext_aks_preview/tests/latest/recordings/test_aks_create_and_update_with_managed_aad_enable_azure_rbac.yaml (hunk 7 above)
Comment (bingosummer) [reply src/aks-preview/azext_aks_preview/tests/latest/recordings/test_aks_create_and_update_with_managed_aad_enable_azure_rbac.yaml:56]: they have been redacted.
https://github.com/Azure/azure-cli-extensions/tree/master/src/aks-preview/azext_aks_preview/tests/latest/recordings

Comment (Anumita) [reply src/aks-preview/azext_aks_preview/tests/latest/recordings/test_aks_create_and_update_with_managed_aad_enable_azure_rbac.yaml:56]: okay, will do

**Comment by palma21** [comment]
@Anumita are we close to have this merged? Help needed?

**Comment by Anumita** [comment]
> @Anumita are we close to have this merged? Help needed?

Hey, the builds are failing due to some tests failing for all pr right now. I have contacted the poc for the tests. Once that is fixed , i think the pr needs to be approved by gtracer or zquingqing1 and then can be merged.

File: src/aks-preview/azext_aks_preview/tests/latest/recordings/test_aks_byo_appgw_with_ingress_appgw_addon.yaml [hunk 9]
```diff
@@ -1790,44 +1825,62 @@ interactions:
...
+      x-aspnet-version:
+      - 4.0.30319
+      x-ms-dirapi-data-contract-version:
+      - '1.6'
+      x-powered-by:
+      - ASP.NET
     status:
-      code: 200
-      message: OK
+      code: 400
```
Comment (bingosummer) [inline src/aks-preview/azext_aks_preview/tests/latest/recordings/test_aks_byo_appgw_with_ingress_appgw_addon.yaml:1882]: Is "bad request" expected?

Comment (Anumita) [reply src/aks-preview/azext_aks_preview/tests/latest/recordings/test_aks_byo_appgw_with_ingress_appgw_addon.yaml:1839]: not sure, i ran the tests again just to be sure, but getting the same

**Pull request merged**